src/components/Secrets/SecretsListView/SecretsListRowWithComponents.tsx

```diff
@@ -0,0 +1,64 @@
+import * as React from 'react';
+import { Label, Spinner, Tooltip, TooltipPosition } from '@patternfly/react-core';
+import { useLinkedServiceAccounts } from '~/hooks/useLinkedServiceAccounts';
+import { useNamespace } from '~/shared/providers/Namespace';
+import { RowFunctionArgs, TableData } from '../../../shared';
+import ActionMenu from '../../../shared/components/action-menu/ActionMenu';
+import { SecretKind } from '../../../types/secret';
+import { useSecretActions } from '../secret-actions';
+import { getSecretRowLabels, getSecretTypetoLabel } from '../utils/secret-utils';
+import { isLinkableSecret } from '../utils/service-account-utils';
+import { secretsTableColumnClasses } from './SecretsListHeaderWithComponents';
+import './SecretsListRowTooltip.scss';
+
+type SecretsListRowProps = RowFunctionArgs<SecretKind>;
+
+const SecretsListRowWithComponents: React.FC<React.PropsWithChildren<SecretsListRowProps>> = ({
+  obj,
+}) => {
+  const actions = useSecretActions(obj);
+  const namespace = useNamespace();
+
+  const { secretLabels } = getSecretRowLabels(obj);
+  const labels =
+    secretLabels !== '-'
+      ? secretLabels.split(', ').map((s) => <Label key={s}>{s}</Label>)
+      : secretLabels;
+
+  const { linkedServiceAccounts, isLoading, error } = useLinkedServiceAccounts(
+    obj.metadata?.namespace || namespace,
+    obj.metadata.name,
+    true,
+  );
+
+  return (
+    <>
+      <TableData className={secretsTableColumnClasses.secretType}>
+        {getSecretTypetoLabel(obj) ?? '-'}
+      </TableData>
+      <TableData className={secretsTableColumnClasses.name}>{obj.metadata?.name}</TableData>
+      <TableData className={secretsTableColumnClasses.components}>
+        <div data-test="components-content">
```

Comment:
dont need a `div` here. add data-test on the TableData

Reply:
would update.

Reply:
done